### ISSUE TITLE
[BUG] `device.register()` get stuck when the WebSocket fails to connect

### ISSUE THREAD

**Issue by kamalbennani** [twilio/twilio-voice.js#231]
- [x] I have verified that the issue occurs with the latest twilio.js release and is not marked as a known issue in the [CHANGELOG.md](https://github.com/twilio/twilio-voice.js/blob/master/CHANGELOG.md).
 - [x] I reviewed the [Common Issues](https://github.com/twilio/twilio-voice.js/blob/master/COMMON_ISSUES.md) and open GitHub issues and verified that this report represents a potentially new issue.
 - [x] I verified that the [Quickstart](https://github.com/TwilioDevEd/voice-javascript-sdk-quickstart-node) application works in my environment.
 - [x] I am not sharing any [Personally Identifiable Information (PII)](https://www.twilio.com/docs/glossary/what-is-personally-identifiable-information-pii)
       or sensitive account information (API keys, credentials, etc.) when reporting this issue.

**Code to reproduce the issue:**

While attempting to initialize a Twilio Device with an expired token results in the failure of executing subsequent code following the invocation of `await device.register()`.
This occurs due to the Twilio WebSocket being closed caused by the token expiration, preventing the `register()` promise to be resolve/reject.

```js
(async () => {
  try {
    const device = new Device('AnyExpiredToken');
    await device.register()
  } finally {
    console.log('log never reached');
  }
})()
```

**Expected behavior:**

The `.register()` function should throw an error indicating that the device has failed to register.

**Actual behavior:**

`register()` promise never resolves.

**Software versions:**

 - [x] Browser(s): 120.0.6099.109
 - [x] Operating System: MacOS 13.5.2 (22G91)
 - [x] twilio.js: 2.8.0
 - [x] Third-party libraries (e.g., Angular, React, etc.): React

**Comment by charliesantos** [comment]
Good find @kamalbennani ! Thanks for submitting. I created an internal ticket to have the associated PR reviewed and tested further.

**Comment by charliesantos** [comment]
fixed in 2.10.1
